### PR TITLE
dialects: (riscv) remove noisy () -> in get_(float_)register and li

### PR DESCRIPTION
**Pull request by superlopuh**
I find this much easier to parse personally. It's a big change, but it's automated, I just ran three find and replace commands on the project

Find and replace:

```
# li
riscv.li (.*) : \(\) -> !riscv.reg
riscv.li ${1} : \(\) -> !riscv.reg
# get_register
riscv.get_register : () -> !riscv.reg
riscv.get_register : !riscv.reg
# get_float_register
riscv.get_float_register : () -> !riscv.freg
riscv.get_float_register : !riscv.freg
```

### CODE REVIEW

File: tests/filecheck/projects/riscv-backend-paper/pres.mlir [hunk 1]
```diff
@@ -71,8 +71,8 @@ riscv_func.func @pres_4(%X : !riscv.reg<a0>, %Y : !riscv.reg<a1>, %Z : !riscv.re
     %zero_5 = riscv_snitch.scfgwi %Z, 898 : (!riscv.reg<a2>) -> !riscv.reg<zero>
     %zero_6 = riscv.csrrsi 1984, 1 : () -> !riscv.reg<zero>
     %i = riscv.addi %zero, 63 : (!riscv.reg<zero>) -> !riscv.reg<a0>
-    %x = riscv.get_float_register() : () -> !riscv.freg<ft0>
```
Comment (superlopuh) [inline tests/filecheck/projects/riscv-backend-paper/pres.mlir:74]: it's weird that we allow this... I don't feel like fixing the whole custom syntax for riscv right now but might be good to look into later

Comment (tobiasgrosser) [reply tests/filecheck/projects/riscv-backend-paper/pres.mlir:74]: Not sure what you believe is weird here?

Comment (superlopuh) [reply tests/filecheck/projects/riscv-backend-paper/pres.mlir:74]: `riscv.get_float_register()` this is an operation that cannot have operands, so allowing the `()` at the end seems off, I would expect the parser to warn the user.

**Comment by tobiasgrosser** [comment]
This looks good to me.

**Review by nazavode — APPROVED** [review]

**Review by PapyChacal — APPROVED** [review]

**Pull request merged**